File: src/main/java/space/davids_digital/neurobots/geom/Line.java
package space.davids_digital.neurobots.geom;

import java.awt.*;

public class Line {
    private DoublePoint pointA;
    private DoublePoint pointB;

    public Line(DoublePoint pointA, DoublePoint pointB) {
        this.pointA = pointA;
        this.pointB = pointB;
    }

    public Line(Point pointA, Point pointB) {
        this.pointA = new DoublePoint(pointA.x, pointA.y);
        this.pointB = new DoublePoint(pointB.x, pointB.y);
    }

    public DoublePoint getPointA() {
        return pointA;
    }

    public void setPointA(DoublePoint pointA) {
        this.pointA = pointA;
    }

    public DoublePoint getPointB() {
        return pointB;
    }

    public void setPointB(DoublePoint pointB) {
        this.pointB = pointB;
    }
}

File: src/main/java/space/davids_digital/neurobots/geom/Line.kt
package space.davids_digital.neurobots.geom

class Line (var pointA: DoublePoint, var pointB: DoublePoint) {
    constructor(ax: Number, ay: Number, bx: Number, by: Number): this(DoublePoint(ax, ay), DoublePoint(bx, by))
    val middle get() = DoublePoint((pointA.x + pointB.x)/2, (pointA.y + pointB.y) / 2)
    val length get() = pointA.distance(pointB)
}